baselines/niid_bench/README.md

```diff
@@ -0,0 +1,131 @@
+| FMNIST | 10 | 10 | sort and partition | 3 labels per client |
+
+
+****Training Hyperparameters:**** There are four FL algorithms and they have many common hyperparameters and few different ones. The following table shows the common hyperparameters and their default values.
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 10 |
+| clients per round | 10 |
+| number of rounds | 50 |
+| number of local epochs | 10 |
+| client resources | {'num_cpus': 4.0, 'num_gpus': 0.0 }|
+| dataset name | cifar10 
+| data partition | Dirichlet (0.5) |
+| batch size | 64 |
+| momentum for SGD | 0.9 |
+
+For FedProx algorithm the proximal parameter is tuned from values {0.001, 0.01, 0.1, 1.0} in the experiments. The default value is 0.01. 
+
+
+## Environment Setup
+
+```bash
+# Setup the base poetry enviroment from the niid_bench directory
+poetry install
+
+# Start the shell
+poetry shell
+
+#  Install tqdm, torch, and torchvision using pip
+pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 --extra-index-url https://download.pytorch.org/whl/cu113
+pip install tqdm
+```
+
+
+## Running the Experiments
+You can run four algorithms fedavg, scaffold, fedprox, and fednova. To run any of them, use any of the corresponding main files. For instance, the following command will run with the default config provided in the corresponding configuration files.
+
+```bash
+python -m niid_bench.main_fedprox
+```
+
+To change the configuration such as dataset or hyperparameters, specify them as part of the command line arguments.
+
+```bash
+python -m niid_bench.main_scaffold dataset_name=mnist partitioning=iid # iid
+python -m niid_bench.main_fedavg dataset_name=mnist partitioning=dirichlet # dirichlet
+python -m niid_bench.main_fednova dataset_name=mnist partitioning=label_quantity labels_per_client=3 # sort and partition
+```
+
+
+## Expected Results
+
+We provide the bash script run_all.sh that can be used to run all configurations for one seed with 7 configurations running at the same time.
+
+```bash
+./run_all.sh
+```
+
+The above command generates results that can be parsed to get the accuracies for each configuration which can be presented in a table (similar to Table 3 in the paper).
+
+| Dataset | partitioning method | FedAvg | SCAFFOLD | FedProx | FedNova |
+| :------ | :------ | :---: | :---: | :---: | :---: |
+| MNIST | IID | | | | |
+| MNIST | Dirichlet (0.5) | | | | |
+| MNIST | Sort and Partition (1) | | | | |
+| MNIST | Sort and Partition (2) | | | | |
+| MNIST | Sort and Partition (3) | | | | |
+| FMNIST | IID | | | | |
+| FMNIST | Dirichlet (0.5) | | | | |
+| FMNIST | Sort and Partition (1) | | | | |
+| FMNIST | Sort and Partition (2) | | | | |
+| FMNIST | Sort and Partition (3) | | | | |
+| CIFAR10 | IID | | | | |
+| CIFAR10 | Dirichlet (0.5) | | | | |
+| CIFAR10 | Sort and Partition (1) | | | | |
+| CIFAR10 | Sort and Partition (2) | | | | |
+| CIFAR10 | Sort and Partition (3) | | | | |
```

Comment:
```suggestion
| Dataset | partitioning method | FedAvg | SCAFFOLD | FedProx | FedNova |
| :------ | :------ | :---: | :---: | :---: | :---: |
| MNIST | IID<br>Dirichlet (0.5)<br>Sort and Partition (1)<br>Sort and Partition (2)<br>Sort and Partition (3) | | | | |
| FMNIST | IID<br>Dirichlet (0.5)<br>Sort and Partition (1)<br>Sort and Partition (2)<br>Sort and Partition (3) | | | | |
| CIFAR10 | IID<br>Dirichlet (0.5)<br>Sort and Partition (1)<br>Sort and Partition (2)<br>Sort and Partition (3) | | | | |
```

Reply:
You can use `<br>` to have something similar to latex's `multi-row`. You could use this to make the table in the "Experiment Setup" section to look a bit nicer if you want.